CT-1912 Allow SAR and ICO Overturned SAR cases to be reassigned

misaka opened the pull request:
SAR and ICO Overturned SAR cases need to be re-assignable to a new team when in
drafting. This change adds that, and removes the same re-assignability from
pending_dacu_disclosure and awaiting_dispatch states, as that wouldn't have
worked anyway and we don't yet know what to do when that happens. Once the case
is accepted by the new team, should it then go back to the state it was in
previously? Probably, but that work is outside the scope of this bug.

Other noteworthy changes in this commit:

- Change permit_trigger_event matcher to allow the 'transition_to' option to be
  specified. When present, the matcher will only match the case state and user
  to the event if the event has a transition_to to the specified state. This
  does mean that events that transition to another/no state will be treated as
  not-present, so another test will need to be added for those.

  See spec/state_machines/sar_event_spec.rb to see how this is used.

- Cases created by standard setup don't need an identifier set, that'll be taken
  from the name of the case as provided in the StandardSetup config.

stephenrichards commented on app/models/team.rb:
```diff
@@ -102,6 +102,10 @@ def active_users
+    else
+      children.active.any?
+    end
   end
```
I don't   like this conflation of children and users.  I think there should be a has_active_children? method on Team, and a has_active_users? method on BusinessUnit.

If  you really need one method here, then call it something that is more descriptive of what it does, and is not misleading, e.g. deleteable?

misaka committed:
CT-1912 Allow SAR and ICO Overturned SAR cases to be reassigned

SAR and ICO Overturned SAR cases need to be re-assignable to a new time when in
drafting. This change adds that, and removes the same re-assignability from
pending_dacu_disclosure and awaiting_dispatch states, as that wouldn't have
worked anyway and we don't yet know what to do when that happens. Once the case
is accepted by the new team, should it then go back to the state it was in
previously? Probably, but that work is outside the scope of this bug.

Other noteworthy changes in this commit:

- Change permit_trigger_event matcher to allow the 'transition_to' option to be
  specified. When present, the matcher will only match the case state and user
  to the event if the event has a transition_to to the specified state. This
  does mean that events that transition to another/no state will be treated as
  not-present, so another test will need to be added for those.

  See spec/state_machines/sar_event_spec.rb to see how this is used.

- Cases created by standard setup don't need an identifier set, that'll be taken
  from the name of the case as provided in the StandardSetup config.

misaka committed:
CT-1912 set creation_date back to 4.business_days.ago

misaka committed:
CT-1912 contract fixture names for ICO Overturned SAR
